`terraform modules` command (#35884)
hashicorp/terraform: website/docs/cli/commands/modules.mdx

---
page_title: 'Command: modules'
description: >-

>   The terraform modules command prints information about all the declared

Reviewer: What kind of information?

>   The terraform modules command prints information about all the declared

Reviewer: What kind of information?

  modules in Terraform configuration.
---

# Command: modules

The `terraform modules` command provides a holistic view of all Terraform
modules and their resolved versions that have been declared in Terraform
configuration for the current working directory.

> 

Reviewer: Can you add context on _when_ someone would want to run this? Could be helpful to touch on using this instead of `modules.json`.

## Usage

> Usage: `terraform modules -json`

Reviewer: What happens if you omit the `-json` flag? Does it error? Or just provide json anyways? This is just me being curious 😅

---
page_title: 'Command: modules'
description: >-
  The terraform modules command prints source and version data about all declared
  modules in Terraform configuration.
---

# Command: modules

The `terraform modules` command provides a holistic view of all Terraform
modules declared in Terraform configuration for the current working directory.

> This can be useful for auditing or policy purposes on module consumption. The

Reviewer: ```suggestion
This command can be useful for auditing or setting policies on module consumption. The
```

> output will include a list of each declared module's key, source and version.

Reviewer: Small tweak for present tense:
```suggestion
output of `terraform modules` includes a list of each declared module's key, source, and version.
```

> -> `terraform modules` requires **Terraform v1.10.0 or later**.

Reviewer: Little tweak for callout styling:
```suggestion
-> **Note**: The `terraform modules` command requires Terraform v1.10.0 or later.
```

## Usage

> Usage: `terraform modules [options]`

Reviewer: On second look, I think leaving the json flag in makes sense right now, but I love leaving the list below for when we do add more options. 
```suggestion
Usage: `terraform modules -json`
```

The following flags are available:

- `-json` - Displays the module declarations in a machine-readable, JSON format.

> Please note that, at this time, the `-json` flag is a _required_ option. In future releases, this command will be extended to allow for additional options.

Reviewer: ```suggestion
The `-json` flag is required to run the `terraform modules` command. In future releases, we will extend this command to allow additional options.
```

> The output includes a `format_version` key, which as of Terraform 1.10.0 has

Reviewer: ```suggestion
The output of `terraform modules` includes a `format_version` key, which is set to the value `"1.0"` in Terraform 1.10.0. The semantics of this version are:
```

> value `"1.0"`. The semantics of this version are:

Reviewer: ```suggestion
```

> - We will increment the minor version, e.g. `"1.1"`, for backward-compatible

Reviewer: ```suggestion
- For minor versions, such as `"1.1"`, we increment `format_version` for backward-compatible
```

>   changes or additions. Ignore any object properties with unrecognized names to

Reviewer: ```suggestion
  changes or additions. Ignore object properties with unrecognized names to
```

Reviewer: How will folks know what an "unrecognized name" is? Maybe you could provide an example?

  remain forward-compatible with future minor versions.

> - We will increment the major version, e.g. `"2.0"`, for changes that are not

Reviewer: ```suggestion
- For major versions, such as `"2.0"`, we increment `format_version` for any changes that are not
```

  backward-compatible. Reject any input which reports an unsupported major
  version.

We will introduce new major versions only within the bounds of
[the Terraform 1.0 Compatibility Promises](/terraform/language/v1-compatibility-promises).

## Output Format

> The following section describes the JSON output format by example, using a pseudo-JSON notation.

Reviewer: ```suggestion
The following example is a representation of the JSON output that `terraform modules -json` returns.
```
